cns/fsnotify/fsnotify.go

```diff
@@ -68,7 +79,14 @@ func (w *watcher) releaseAll(ctx context.Context) {
 		}
 		file.Close()
 		podInterfaceID := string(data)
-
+		// in case of stateless CNI for Windows, CNS needs to remove HNS endpoitns first
+		if isStalessCNIMode(w.cnsconfig) {
+			// remove HNS endpoint
+			w.log.Info("deleting HNS Endpoint asynchronously")
+			if err := w.deleteHNSEndpoint(ctx, containerID); err != nil {
```

Comment:
this is common file. should we name it as `deleteEndpoint` and move hns api to windows specific file?

Reply:
i see you are doing it. probably rename `deleteEndpoint` in common file

Reply:
Make sense